### PR TITLE
enable HPO for OTE_cli (brave_new_world)

### PR DESCRIPTION
**Pull request by e-ddykim**
This PR is for enabling HPO for OTE_cli (brave_new_world).
The list of changes are as follows:

* 'hpo_config.yaml' file are added for each model.
* Some codes are changed to pass the validation score every epoch via OTE_SDK's standard API.

### CODE REVIEW

**Comment by sovrasov** [comment]
@e-ddykim please retarget this PR to `vs/nous` branch and rebase.

**Comment by e-ddykim** [comment]
> @e-ddykim please retarget this PR to `vs/nous` branch and rebase.

@sovrasov I rebased this PR to `vs/nous`.

**Comment by sovrasov** [comment]
Ok, I've also retargeted the PR.

File: torchreid/integration/sc/task.py [hunk 1]
```diff
@@ -268,6 +268,7 @@ def train(self, dataset: Dataset, output_model: ModelEntity, train_parameters: O
         else:
             train_model = deepcopy(self._model)
 
+        self._cfg.train.lr = self._hyperparams.learning_parameters.learning_rate
```
Comment (sovrasov) [inline torchreid/integration/sc/task.py:271]: All the hyperparameters should be declared and documented in `torchreid/integration/sc/parameters.py` and `torchreid/integration/sc/configuration.yaml `. At the moment I don't see `learning_rate` among the declared papameters.

File: torchreid/integration/sc/utils.py [hunk 2]
```diff
@@ -294,7 +295,18 @@ def __init__(self, update_progress_callback: UpdateProgressCallback, **kwargs):
 
     def on_train_batch_end(self, batch, logs=None):
         super().on_train_batch_end(batch, logs)
-        self.update_progress_callback(self.get_progress())
+        self.update_progress_callback(self.get_progress(), logs)
+
+    def on_epoch_end(self, epoch, logs=None):
+        self.past_epoch_duration.append(time.time() - self.start_epoch_time)
+        self.__calculate_average_epoch()
+        self.update_progress_callback(self.get_progress(), logs)
```
Comment (sovrasov) [inline torchreid/integration/sc/utils.py:303]: `update_progress_callback` should implement the following signature: `def __call__(self, progress: float, score: Optional[float] = None):` (see UpdateProgressCallback in ote_sdk). 
`logs` argument is not expected at the moment, so these changes are invalid from the sc/ote sdk perspective. In the SC environment such call could imply training container failure. I'd propose you making modification of the sdk itself in the impt repo, before making these changes here.

**Review by sovrasov — CHANGES_REQUESTED** [review]

File: torchreid/integration/sc/utils.py (hunk 2 above)
Comment (sovrasov) [reply torchreid/integration/sc/utils.py:303]: Please disregard the last comment. It's better to rename the `logs` argument to avoid confusion.

File: torchreid/integration/sc/task.py (hunk 1 above)
Comment (e-ddykim) [reply torchreid/integration/sc/task.py:271]: I added `learning_rate` in `torchreid/integration/sc/parameters.py` as you reviewed. But there is already `learning_rate` in `torchreid/integration/sc/configuration.yaml@L45`. Thank you!

File: torchreid/integration/sc/utils.py (hunk 2 above)
Comment (e-ddykim) [reply torchreid/integration/sc/utils.py:303]: The `on_epoch_end` is overriding `ote_sdk.usecases.reporting.callback.Callback.on_epoch_end(self, epoch, logs=None)`. Even though renaming `logs` to `score` is clearer, pylint complains about this as (W0237: arguments-renamed).

**Review by sovrasov — APPROVED** [review]
Overall, this PR breaks the multilabel classification usecase: we are using two different configs in one template (for multiclass and for multilabel). Each config has it's own default learning rate value and lr_finder strategy, and classification task switches between these configs internally depending on the given label schema. Now HPO is responsible for learning rate tuning, and this parameter is exposed via task ConfigurableParameters mechanism. Current HPO config seems to be totally unsuitable for the multilabel classification. The next steps here are subject for the future discussion.
